include/Hdlcpp.hpp

```diff
@@ -86,19 +89,30 @@ class Buffer {
 };
 
 //! @param Capacity The buffer size to be allocated for encoding/decoding frames
-template<size_t Capacity>
 class Hdlcpp {
-    static_assert(Capacity > 0, "HDLCPP requires a buffer size larger than 0");
-    using Container = std::array<uint8_t, Capacity>;
+    using Container = std::span<uint8_t>;
 public:
+    template<size_t Capacity>
+    using StaticBuffer = std::array<Container::value_type, Capacity>;
+
+    template<size_t Capacity>
+    struct Calculate
+    {
+        // For details see: https://en.wikipedia.org/wiki/High-Level_Data_Link_Control#Structure
+        static constexpr size_t Overhead = Capacity * 2 + 8;
```

Comment:
IMO the names Capacity and Overhead are a bit misleading. Capacity is the original data size and Overhead is the encoded size (I would assume: Overhead = encoded size - original data size).

Reply:
I agree that the names are not the best :disappointed: 
what about 
```cpp
static constexpr size_t WithOverhead = ...
```

Then creating a buffer would be like this
```cpp
static Hdlcpp::Hdlcpp::StaticBuffer<Hdlcpp::Hdlcpp::Calculate<HostProtocol::ReadBufferSize>::WithOverhead> readBuffer{};
```

Reply:
These helpers should probably be moved out of the class so it is just scoped by the namespace
```cpp
static Hdlcpp::StaticBuffer<Hdlcpp::Calculate<HostProtocol::ReadBufferSize>::WithOverhead> readBuffer{};
```
Which makes it a bit more readable

Reply:
That works